### DISCUSSION TITLE
Unicode 16 Semigraphical Characters

### DISCUSSION THREAD

**Discussion by EnronEvolved** [ghostty-org/ghostty#4108] [Feature Requests, Ideas] [answered]
Do plans exist to add the following ranges of characters to the built-in graphics?

* The new block elements [U+1FBCE, U+1FBCF], box drawing characters [U+1FBD0, U+1FBE0), and justified block and circle elements [U+1FBE0, U+1FBF0) in Symbols for Legacy Computing.
* The separated block sextants [U+1CE51, U+1CE90) and quadrants [U+1CC21, U+1CC30).
* The block octants [U+1CD00, U+1CDE5].
* The circle segments [U+1CC30, U+1CC40)
* Any other new characters in the recently-standardised Symbols for Legacy Computing Supplement block.

I'd argue that a significant number of the characters in this block, mainly those in the aforementioned ranges, are in scope.

**Comment by JakeWharton** [comment]
See https://github.com/ghostty-org/ghostty/pull/2439 and https://github.com/ghostty-org/ghostty/pull/2975 for some/most.

**Comment by EnronEvolved** [comment]
Upon a more detailed review (prompted by JakeWharton), I've found the following ranges which render as object replacements on 1.0.1-main+a8e5eef1, and are simple enough to to be in scope: 

- [ ] The separated block sextants.
- [ ] The block octants.
- [ ] The sixteenth blocks and their composites [U+1CE90, U+1CEB0) 
- [ ] The aforementioned circle segments.
- [ ] Several short ranges of additional box drawings: [U+1CC09, U+1CC0A], [U+1CC1B, U+1CC20], and [U+1CE16, U+1CE19].
- [ ] U+1CC8E and U+1CC8F, "Left and Upper One Eighth Block Containing Black Small Square" and its inverse (respectively).

There are also a few more unconventional characters that _might_ warrant special rendering (in my rough descending order of suitability):

* The left and right ellipse characters U+1CE0B and U+1CE0C, and the dashed line components U+1CE0D and U+1CE0E.
* The lines with tick marks, [U+1CC90, U+1CC91] and [U+1CC0F, U+1CE15], and ruler segments [U+1CC05, U+1CC07].
* The geometrically-patterned fill characters [U+1CC40, U+1CC45].
* The large type characters [U+1CE1A, U+1CE50].
* The schematic symbols ("designed to be used with box drawing characters") [U+1CC08, U+1CC1A].

I also note that the separated quadrants, as they currently render, look a little fuzzy compared to the other block elements.

**Comment by pluiedev** [comment, marked as answer]
As of b0d9b0dee0ed34b2732933a29bfd173290c2577b, these have all been implemented:

- [x] All Symbols for Legacy Computing (U+1FB00..FF)
- [x] Separated block sextants (U+1CE51..8F)
- [x] Separated block quadrants (U+1CC21..2F)
- [x] Octants (U+1CD00..E5)
- [x] Circle segments (U+1CC30..3F)
- [x] Sixteenth blocks (U+1CE90..AF)
- [x] Box drawing corners (U+1CC1B..1E, U+1CE16..19)
- [x] Left/right white half-ellipses (U+1CE0B..0C)

All and all I think the vast majority of the symbols mentioned here are covered. Feel free to open a new discussion if you really need the other symbols as well
